deploy.md

```diff
@@ -1,59 +1,142 @@
+
+# Pre-deploy
+
+In this stage, we prep everything for the release in the git repo.
+
+## Update code and packages
 
 - ensure you have the latest version of the code
 
-`git pull`
+  ```
+  git checkout release
+  git pull upstream release
+  ```
 
 - update/install any packages
 
-`npm install`
+  ```
+  npm install
+  ```
 
-# Update version numbers
+## Update CHANGELOG and version numbers
 - set a new version of the addon by running npm version
 
-`npm version patch`
-
-- update the changelog:
+  ```
+  npm version patch
+  ```
 
-`./node_modules/.bin/conventional-changelog -i CHANGELOG.md -s -r 0`
+- update the changelog
 
-`git add CHANGELOG.md`
+  ```
+  ./node_modules/.bin/conventional-changelog -i CHANGELOG.md -s -r 0
+  git add CHANGELOG.md
+  git commit -m '1.X.Y changelog' # where X and Y are the minor and patch version you just upgraded to
+  ```
 
-`git commit -m '1.0.X changelog' # where X is the patch version you just upgraded to`
+## Update repository
 
-- update remote repository
+  ```
+  # merge changes to master
+  git checkout master
+  git merge release
+  git push upstream master
 
-`git push origin master`
+  # note that you might want to cherry-pick commits from the release branch if there exists some commits
+  # that are not supposed to be merged to master, for example, bugfixes for the deprecated features in the master branch.
+  git checkout master
+  git cherry-pick some_commit_hash
+  git push upstream master
 
-`git push origin --tags`
+  # push the release branch to remote
+  git push upstream release
+  git push upstream --tags
+  ```
 
 # Deploy
 
-- create / instantiate virtual env. Note the path name. it is important so that the python virtual env doesn't get included in the addon.
+We use `fabric` in a Python virtual environment to do the job,
 
-`virtualenv activity-streams-env`
+- create / instantiate virtual env. Note the path name, it is important so that the python virtual env doesn't get included in the addon
 
-`source ./activity-streams-env/bin/activate`
+  If it is the __first__ time to deploy on your machine,
+  ```
+  virtualenv activity-streams-env
+  source ./activity-streams-env/bin/activate
+  pip install -r requirements.txt
+  ```
 
-`pip install -r requirements.txt`
+  Otherwise,
+  ```
+  source ./activity-streams-env/bin/activate
+  ```
 
 - deploy
 
-Important: Ensure there are no extra files in the directory. Any file that is not in .jpmignore will be included in the addon.
-Make extra sure the .amo_config.json is named properly, otherwise it might be included in the addon XPI
+  We will deploy off the `release` branch,
+
+  ```
+  git checkout release
+  ```
+
+  Important: Ensure there are no extra files in the directory. Any file that is not in .jpmignore will be included in the addon. Make extra sure that the `.amo_config.json` is named properly, otherwise it might be included in the addon XPI.
```

Comment:
Ultra nit: I know this is just copy-pasta, but not sure if we want to change these from "addon" to "add-on" (since we use the hyphenated version elsewhere -- and it's hyphenated at https://github.com/mozilla/addons-linter).